nomad/vault.go

```diff
@@ -992,6 +992,20 @@ func (v *vaultClient) CreateToken(ctx context.Context, a *structs.Allocation, ta
 		namespaceForTask = taskVault.Namespace
 	}
 
+	// Set entity alias if given
+	entityAlias := ""
+
+	// If set in the server config, use an entity_alias
+	if v.config.EntityAlias != "" {
+		entityAlias = taskVault.EntityAlias
+	}
+
+	// If set at the task level, use the task-specific alias
+	// even if the server config has an entity_alias as well
+	if taskVault.EntityAlias != "" {
+		entityAlias = taskVault.EntityAlias
+	}
```

Comment:
Oh yeah - the first one should be setting entityAlias to v.config.EntityAlias but I copy/pasted wrong

Changed in the latest revision - with a slight logical change so that the request doesnt unnecessarily include the EntityAlias key